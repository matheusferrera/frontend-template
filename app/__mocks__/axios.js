/* eslint-disable no-undef */
import { loginMockData, registerMockData, userMockData } from "../test/mockData";

// Cria um objeto simulado que atua como uma vers√£o personalizada do Axios
const customAxios = {
  mockLogin: jest.fn().mockResolvedValue(loginMockData),
  mockRegister: jest.fn().mockResolvedValue(registerMockData),
  mockUser: jest.fn().mockResolvedValue(userMockData),
  mockDate: jest.fn().mockResolvedValue(new Date()),
};

export default customAxios;
