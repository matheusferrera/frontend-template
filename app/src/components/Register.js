import React, { useState, useRef } from "react";
import Form from "react-validation/build/form";
import Input from "react-validation/build/input";
import CheckButton from "react-validation/build/button";
import { isEmail } from "validator";

import { useAuth } from "../contexts/AuthContext";

const required = value => {
  if (!value) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        This field is required!
      </div>
    );
  }
};

const validEmail = value => {
  if (!isEmail(value)) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        This is not a valid email.
      </div>
    );
  }
};

const vusername = value => {
  if (value.length < 3 || value.length > 20) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        The username must be between 3 and 20 characters.
      </div>
    );
  }
};

const vpassword = value => {
  if (value.length < 6 || value.length > 40) {
    return (
      <div
        className="my-1 alert alert-danger"
        role="alert"
      >
        The password must be between 6 and 40 characters.
      </div>
    );
  }
};

const Register = () => {
  const form = useRef();
  const checkBtn = useRef();

  const [email, setEmail] = useState("");
  const [name, setName] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [passwordConfirmation, setPasswordConfirmation] = useState("");
  const [successful, setSuccessful] = useState(false);
  const [loading, setLoading] = useState(false);
  const [message, setMessage] = useState("");

  const { register } = useAuth();

  const onChangeEmail = e => {
    const email = e.target.value;
    setEmail(email);
  };

  const onChangeName = e => {
    const name = e.target.value;
    setName(name);
  };

  const onChangeUsername = e => {
    const username = e.target.value;
    setUsername(username);
  };

  const onChangePassword = e => {
    const password = e.target.value;
    setPassword(password);
  };

  const onChangePasswordConfirmation = e => {
    const passwordConfirmation = e.target.value;
    setPasswordConfirmation(passwordConfirmation);
  };

  const handleRegister = async e => {
    e.preventDefault();

    setMessage("");
    setLoading(true);
    setSuccessful(false);

    form.current.validateAll();

    if (password !== passwordConfirmation) {
      setMessage("Password confirmation does not match the password.");
      setSuccessful(false);
      setLoading(false);
      return;
    }

    register(email, name, username, password, passwordConfirmation)
      .then(response => {
        setMessage(response);
        setSuccessful(true);
        setLoading(false);
      })
      .catch(error => {
        const resMessage = (error.response && error.response.data && error.response.data.message) || error.message || error.toString();
        setMessage(resMessage);
        setSuccessful(false);
        setLoading(false);
      });
  };

  return (
    <div className="col-md-12">
      <div className="card card-container">
        <img
          src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
          alt="profile-img"
          className="profile-img-card"
        />

        <Form
          onSubmit={handleRegister}
          ref={form}
        >
          {!successful && (
            <div>
              <div className="mb-3">
                <label
                  htmlFor="email"
                  className="form-label"
                >
                  Email
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="email"
                  value={email}
                  onChange={onChangeEmail}
                  validations={[required, validEmail]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="name"
                  className="form-label"
                >
                  Name
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="name"
                  value={name}
                  onChange={onChangeName}
                  validations={[required]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="username"
                  className="form-label"
                >
                  Username
                </label>
                <Input
                  type="text"
                  className="form-control"
                  name="username"
                  value={username}
                  onChange={onChangeUsername}
                  validations={[required, vusername]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="password"
                  className="form-label"
                >
                  Password
                </label>
                <Input
                  type="password"
                  className="form-control"
                  name="password"
                  value={password}
                  onChange={onChangePassword}
                  validations={[required, vpassword]}
                />
              </div>

              <div className="mb-3">
                <label
                  htmlFor="passwordConfirmation"
                  className="form-label"
                >
                  Password Confirmation
                </label>
                <Input
                  type="password"
                  className="form-control"
                  name="passwordConfirmation"
                  value={passwordConfirmation}
                  onChange={onChangePasswordConfirmation}
                  validations={[required, vpassword]}
                />
              </div>

              <div className="mb-3">
                <button
                  className="btn btn-primary btn-block"
                  disabled={loading}
                >
                  {loading && <span className="spinner-border spinner-border-sm"></span>}
                  <span>Sign Up </span>
                </button>
              </div>
            </div>
          )}

          {message && (
            <div className="mb-3">
              <div
                className={successful ? "alert alert-success" : "my-1 alert alert-danger"}
                role="alert"
              >
                {message}
              </div>
            </div>
          )}
          <CheckButton
            style={{ display: "none" }}
            ref={checkBtn}
          />
        </Form>
      </div>
    </div>
  );
};

export default Register;
