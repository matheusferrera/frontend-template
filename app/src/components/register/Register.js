import React from "react";
import { useState } from "react";
import { useNavigate } from "react-router-dom";

import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Card from "@mui/material/Card";
import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";

import { useAuth } from "../../contexts/AuthContext";
import AvisoDePrivacidadeModal from "../modals/AvisoDePrivacidadeModal";
import TermoDeUsoModal from "../modals/TermoDeUsoModal";
import RegisterForm from "./RegisterForm";

const Register = () => {
  const { register } = useAuth();
  const navigate = useNavigate();

  const [loading, setLoading] = useState(false);

  const recaptchaRef = React.createRef();

  const [termoDeUsoModal, setTermoDeUsoModal] = useState(false);
  const [avisoDePrivacidadeModal, setAvisoDePrivacidadeModal] = useState(false);

  const handleTermoDeUsoShow = () => setTermoDeUsoModal(true);
  const handleTermoDeUsoClose = () => setTermoDeUsoModal(false);

  const handleAvisoDePrivacidadeShow = () => setAvisoDePrivacidadeModal(true);
  const handleAvisoDePrivacidadeClose = () => setAvisoDePrivacidadeModal(false);

  const handleSubmit = (values, { setSubmitting }) => {
    const token = recaptchaRef.current.getValue();

    setLoading(true);

    if (!token) {
      alert("Por favor, confirme que você não é um robô");
      setLoading(false);
      setSubmitting(false);
      return;
    }

    register(values.email, values.name, values.password, values.passwordConfirmation)
      .then(response => {
        navigate("/login");
        alert(response);
      })
      .catch(error => {
        const resMessage = (error.response && error.response.data && error.response.data.message) || error.message || error.toString();
        alert(resMessage);
      })
      .finally(() => {
        setLoading(false);
        setSubmitting(false);
      });
  };

  return (
    <Box
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        minHeight: "100vh",
      }}
    >
      <Stack
        alignItems="center"
        justifyContent="center"
        sx={{ height: 1 }}
      >
        <Card
          sx={{
            p: 5,
            width: 1,
            maxWidth: 625,
            height: 1,
            maxHeight: 900,
          }}
        >
          <Typography variant="h3">Registrar-se</Typography>

          <Typography
            variant="body2"
            sx={{ mt: 2, mb: 3, color: "text.grey" }}
          >
            Crie o seu perfil Progredir
          </Typography>

          <RegisterForm
            loading={loading}
            handleSubmit={handleSubmit}
            recaptchaRef={recaptchaRef}
            handleAvisoDePrivacidadeShow={handleAvisoDePrivacidadeShow}
            handleTermoDeUsoShow={handleTermoDeUsoShow}
          />

          <Button
            variant="outlined"
            sx={{ mt: 2 }}
            href="/parceiro_login"
          >
            Já possuo conta
          </Button>

          <Button
            variant="outlined"
            sx={{ mt: 2, ml: 2 }}
            href="/activate"
          >
            Ativar conta
          </Button>

          <TermoDeUsoModal
            showModal={termoDeUsoModal}
            handleClose={handleTermoDeUsoClose}
          />
          <AvisoDePrivacidadeModal
            showModal={avisoDePrivacidadeModal}
            handleClose={handleAvisoDePrivacidadeClose}
          />
        </Card>
      </Stack>
    </Box>
  );
};

export default Register;
