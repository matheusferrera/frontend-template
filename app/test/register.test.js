import React from "react";
import { act } from "react-dom/test-utils";

import { describe, expect, it, jest } from "@jest/globals";
import { fireEvent, render, screen } from "@testing-library/react";
import axios from "axios";

import Register from "../src/components/register/Register";
import { registerMockData } from "./mockData";
import { TestWrapper } from "./testWrapper";

// jest.mock("axios"); // Mockando o módulo axios
jest.mock("../src/components/modals/AvisoDePrivacidadeModal");
jest.mock("../src/components/modals/TermoDeUsoModal");

describe("Teste de registro", () => {
  it("Teste de registro usando mock", async () => {
    render(
      <TestWrapper>
        <Register />
      </TestWrapper>,
    );

    const emailInput = screen.getByLabelText("Email");
    const nameInput = screen.getByLabelText("Nome");
    const passwordInput = screen.getByLabelText("Senha");
    const confirmPasswordInput = screen.getByLabelText("Confirmação de Senha");
    const submitButton = screen.getByText("Registrar");

    // Preencha os campos do formulário
    fireEvent.change(emailInput, { target: { value: "testuser" } });
    fireEvent.change(nameInput, { target: { value: "testname" } });
    fireEvent.change(passwordInput, { target: { value: "testpassword" } });
    fireEvent.change(confirmPasswordInput, { target: { value: "testpassword" } });

    // Envie o formulário
    fireEvent.click(submitButton);

    // Aguarde até que a chamada à API seja concluída
    await act(async () => {
      await axios.mockRegister(registerMockData); // Simula uma chamada à API usando a função `get` do Axios
      expect(axios.mockRegister).toHaveBeenCalledWith({
        data: {
          results: [
            {
              email: "test@email.com",
              name: "Teste",
              username: "teste",
              password: "123456",
            },
          ],
        },
      }); // Verifica se axios.get foi chamado com a URL correta
    });
    expect(axios.mockRegister).toHaveBeenCalledTimes(1); // Verifica se a função `get` do Axios foi chamada uma vez durante o teste
  });
});
