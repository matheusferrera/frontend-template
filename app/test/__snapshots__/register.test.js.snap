// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de registro Register renderizada corretamente 1`] = `
<div
  className="MuiBox-root css-1o0ed15"
>
  <div
    className="MuiStack-root css-ph9haw-MuiStack-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-wjismy-MuiPaper-root-MuiCard-root"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3 css-8jown4-MuiTypography-root"
      >
        Registrar-se
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body2 css-w5whsq-MuiTypography-root"
      >
        Crie o seu perfil Progredir
      </p>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiStack-root css-v2woy3-MuiStack-root"
        >
          <div
            className="MuiFormControl-root MuiTextField-root css-1eq0ymp-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-bmx3g7-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Email
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-rwt5ya-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
              variant="filled"
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input css-andsdq-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root css-1eq0ymp-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-bmx3g7-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="name"
              id="name-label"
            >
              Nome
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-rwt5ya-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
              variant="filled"
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input css-andsdq-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                id="name"
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Nome"
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root css-1eq0ymp-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-bmx3g7-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Senha
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-jundqn-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
              variant="filled"
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd css-9f3xj7-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Senha"
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-10649of-MuiInputAdornment-root"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-17krgaq-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span />
                </button>
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root css-1eq0ymp-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-filled css-bmx3g7-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="passwordConfirmation"
              id="passwordConfirmation-label"
            >
              Confirmação de Senha
            </label>
            <div
              className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-jundqn-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
              variant="filled"
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd css-9f3xj7-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                id="passwordConfirmation"
                name="passwordConfirmation"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Confirmação de Senha"
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-10649of-MuiInputAdornment-root"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-17krgaq-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="mt-2"
        />
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-txgl9-MuiFormControlLabel-root"
        >
          <span
            className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1cu7bhi-MuiButtonBase-root-MuiCheckbox-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate={false}
              id="toggle"
              name="toggle"
              onChange={[Function]}
              required={false}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <p
            className="MuiTypography-root MuiTypography-body1 css-hjr1b1-MuiTypography-root"
          >
            Estou ciente e concordo com o
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Termo de Uso
            </a>
            e
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Aviso de Privacidade
            </a>
          </p>
        </label>
        <button
          className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-1p1ddem-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={false}
          id=":r4:"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Registrar
        </button>
      </form>
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1cbccbr-MuiButtonBase-root-MuiButton-root"
        href="/parceiro_login"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Já possuo conta
      </a>
      <div />
      <div />
    </div>
  </div>
</div>
`;
